clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -1460,6 +1460,15 @@ void Sema::MarkDevice(void) {
   // it is recursive.
   MarkDeviceFunction Marker(*this);
   Marker.SYCLCG.addToCallGraph(getASTContext().getTranslationUnitDecl());
+
+  // Iterate through SYCL_EXTERNAL functions and add them to the device decls.
+  for (const auto &entry : *Marker.SYCLCG.getRoot()) {
```

Comment:
IIRC, @Fznamznon pointed out that this won't add class member functions, which are marked as `SYCL_EXTERNAL` into list of device functions which needs to be checked.
This is not critical, I guess, but I think deserves double-checking and probably a `TODO`, that attribute handling should be improved for class member functions as well

Reply:
I wasn't sure about it, so I double checked it locally, seems everything is fine with class members, but we could add a corresponding test in next patches.